diff -r b681d69af34d tools/osx/make-app-bundle
--- a/tools/osx/make-app-bundle	Tue Jan 08 02:58:45 2013 +0000
+++ b/tools/osx/make-app-bundle	Wed Jan 16 10:36:10 2013 +0900
@@ -29,14 +29,17 @@
 LIB=${MACOS}/lib
 SHARE=${MACOS}/share
 RELEASE=Release
-DMG=rawtherapee_mac32_`date +%F`_`hg -R . branch`_`hg parents --template '{latesttag}i.{latesttagdistance}-{node|short}'`.dmg
 EXECUTABLE=rawtherapee
+RT_VERSION=`awk '/^Version: / { print $2 }' $RELEASE/AboutThisBuild.txt`
+BIT_DEPTH=`awk '/^Bit depth: / { print $3 }' $RELEASE/AboutThisBuild.txt`
+DMG=rawtherapee_mac${BIT_DEPTH}_`date +%F`_`hg -R . branch`_`hg parents --template '{latesttag}i.{latesttagdistance}-{node|short}'`.dmg
 
-#Find where MacPorts is installed.  We take a known binary (cmake), which is in <MacPorts>/bin, and 
-# go up a level to get the main install folder.
-MACPORTS_PREFIX=`which cmake`
-MACPORTS_PREFIX=`dirname $MACPORTS_PREFIX`
-MACPORTS_PREFIX=`dirname $MACPORTS_PREFIX`
+# GTK/MACPORTS PREFIX
+MACPORTS_PREFIX=`otool -L $RELEASE/$EXECUTABLE | awk '/libgtk-.*dylib/ { print $1 }'`
+MACPORTS_PREFIX=${MACPORTS_PREFIX%/lib/*}
+
+# ORIGINAL MACPORTS PREFIX
+#MACPORTS_PREFIX=`which port | sed "s|/bin/port$||"`
 
 if [ ! -d ${RELEASE} ]; then
 	echo "Please run this from the root of the project; i.e. './tools/osx/make-app-bundle'."
@@ -70,10 +73,12 @@
 #Copy over etc files, and modify as needed
 echo "Copying configuration files from ${MACPORTS_PREFIX} and modifying for standalone app bundle..."
 cp -R $MACPORTS_PREFIX/etc/gtk-2.0 ${ETC}
-cp -R $MACPORTS_PREFIX/etc/pango ${ETC}
-ESCAPED_MACPORTS_PREFIX=`echo ${MACPORTS_PREFIX} | sed -e 's/\\//\\\\\\//g'`
-sed -i .bak -e "s/${ESCAPED_MACPORTS_PREFIX}/@executable_path/g" ${ETC}/gtk-2.0/gdk-pixbuf.loaders ${ETC}/gtk-2.0/gtk.immodules ${ETC}/pango/pango.modules
-echo -e "[Pango]\nModuleFiles = /tmp/${EXECUTABLE}_pango.modules" > ${ETC}/pango/pangorc
+cp -R $MACPORTS_PREFIX/etc/pango ${ETC} || mkdir ${ETC}/pango
+$MACPORTS_PREFIX/bin/gtk-query-immodules-2.0 \
+$MACPORTS_PREFIX/lib/gtk-2.0/*/immodules/*.so | sed "s|$MACPORTS_PREFIX|@executable_path|" > $ETC/gtk-2.0/gtk.immodules
+$MACPORTS_PREFIX/bin/gdk-pixbuf-query-loaders | sed "s|$MACPORTS_PREFIX|@executable_path|" > $ETC/gtk-2.0/gdk-pixbuf.loaders
+$MACPORTS_PREFIX/bin/pango-querymodules | sed "s|$MACPORTS_PREFIX|/tmp/$MACOS|" > $ETC/pango/pango.modules
+printf '[Pango]\nModuleFiles = %s' /tmp/$ETC/pango/pango.modules > $ETC/pango/pangorc
 rm ${LIB}/gdk-pixbuf-2.0/2.10.0/loaders.cache
 
 #Copy over the release files
@@ -82,9 +87,36 @@
 
 #Copy application-specific stuff like icons and startup script
 echo "Creating required application bundle files..."
-cp ./tools/osx/Info.plist ${CONTENTS}
 cp tools/osx/Icons.icns ${RESOURCES}
 cp tools/osx/start ${MACOS}
+cat > $CONTENTS/Info.plist <<__EOF__
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple Computer//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>CFBundleDevelopmentRegion</key>
+	<string>English</string>
+	<key>CFBundleExecutable</key>
+	<string>start</string>
+	<key>CFBundleIconFile</key>
+	<string>Icons.icns</string>
+	<key>CFBundleIdentifier</key>
+	<string>com.rawtherapee.rawtherapee</string>
+	<key>CFBundleInfoDictionaryVersion</key>
+	<string>6.0</string>
+	<key>CFBundleName</key>
+	<string>RawTherapee</string>
+	<key>CFBundlePackageType</key>
+	<string>APPL</string>
+	<key>CFBundleShortVersionString</key>
+	<string>${RT_VERSION:0:3}</string>
+	<key>CFBundleSignature</key>
+	<string>APPL</string>
+	<key>CFBundleVersion</key>
+	<string>${RT_VERSION}</string>
+</dict>
+</plist>
+__EOF__
 
 #Copy and relink the explicitly defined libraries
 echo "Recursively copying libraries referenced by executable..."
diff -r b681d69af34d tools/osx/start
--- a/tools/osx/start	Tue Jan 08 02:58:45 2013 +0000
+++ b/tools/osx/start	Wed Jan 16 10:36:10 2013 +0900
@@ -1,19 +1,18 @@
 #!/bin/bash
 
-CWD=`dirname "$0"`
-echo $CWD
+CWD="${0%/*}"
 
 export DYLD_LIBRARY_PATH="${CWD}/lib:$DYLD_LIBRARY_PATH"
 export GTK_DATA_PREFIX="${CWD}"
 export GTK_DATA_DIRS="${CWD}"
 export GTK_EXE_PREFIX="${CWD}"
 export GTK_PATH="${CWD}"
-
 export GTK2_RC_FILES="${CWD}/etc/gtk-2.0/gtkrc"
-export GTK_IM_MODULE_FILE="${CWD}/gtk-2.0/gtk.immodules"
+export GTK_IM_MODULE_FILE="${CWD}/etc/gtk-2.0/gtk.immodules"
 export GDK_PIXBUF_MODULE_FILE="${CWD}/etc/gtk-2.0/gdk-pixbuf.loaders"
 export PANGO_RC_FILE="${CWD}/etc/pango/pangorc"
+export XDG_DATA_DIRS="${CWD}/share:$XDG_DATA_DIRS"
 
-cp "${CWD}/etc/pango/pango.modules" /tmp/rawtherapee_pango.modules
+ln -sf "${0%/Contents/*}" /tmp
 
-"${CWD}/rawtherapee"
+exec "${CWD}/rawtherapee"
